src/nbl/asset/interchange/CGLSLLoader.cpp

```diff
@@ -45,7 +45,7 @@ SAssetBundle CGLSLLoader::loadAsset(system::IFile* _file, const IAssetLoader::SA
 		return {};
 	}
 
-	auto shader = core::make_smart_refctd_ptr<ICPUShader>(reinterpret_cast<char*>(source), found->second, filename.string());
+	auto shader = core::make_smart_refctd_ptr<ICPUShader>(reinterpret_cast<char*>(source), found->second, IShader::ECT_GLSL, filename.string());
```

Comment:
Tasks for @deprilula28 :
1. Maybe we should load to `ICPUBuffer` directly and create the `ICPUShader` from that?
2. Create a `CHLSLLoader` similar to `CGLSLLoader` but:
2a. Only file extensions it associates with are `*.h` and `*.hlsl`
2b. Check loadability by checking that file is under 64MB and regexping the file for any reserved keywords https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-appendix-keywords
2c. it returns a `ICPUShader` not an `ICPUSpecializedShader`

Then I want/need a SPIR-V Loader and DXIL Loader, these will require using SPIR-V Tools and DXC to check loadability (validitity) and check integrity before returnin the asset.

@Erfan-Ahmadi we need a `ESS_LIBRARY` to deal with HLSL as there's no way to infer the stage, unless we start using file extensions like `.vert.hlsl`. Even then we'd still need `ESS_LIBRARY` to deal with stageless bits of HLSL we want to link later.